beacon_node/beacon_chain/src/schema_change/migration_schema_v24.rs

```diff
@@ -0,0 +1,292 @@
+                    ?state_root,
+                    "Ignoring state summary prior to split slot"
+                );
+            } else {
+                // 1. Store snapshot or diff at this slot (if required).
+                let storage_strategy = db.hot_storage_strategy(slot)?;
+                debug!(
+                    %slot,
+                    ?state_root,
+                    ?storage_strategy,
+                    "Migrating state summary"
+                );
+
+                match storage_strategy {
+                    StorageStrategy::DiffFrom(_) | StorageStrategy::Snapshot => {
+                        // We choose to not compute states during the epoch with block replay for
+                        // simplicity. Therefore we can only support a hot heriarchy config where the
+                        // lowest layer is >= 5.
+                        if slot % T::EthSpec::slots_per_epoch() != 0 {
+                            return Err(Error::MigrationError(
+                                "Hot hierarchy config lowest value must be >= 5".to_owned(),
+                            ));
+                        }
+
+                        // Load the full state and re-store it as a snapshot or diff.
+                        let state = get_full_state_v22(&db.hot_db, &state_root, &db.spec)?
+                            .ok_or(Error::MissingState(state_root))?;
+
+                        // Store immediately so that future diffs can load and diff from it.
+                        let mut ops = vec![];
+                        // We must commit the hot state summary immediatelly, otherwise we can't diff
+                        // against it and future writes will fail. That's why we write the new hot
+                        // summaries in a different column to have both new and old data present at
+                        // once. Otherwise if the process crashes during the migration the database will
+                        // be broken.
+                        db.store_hot_state_summary(&state_root, &state, &mut ops)?;
+                        db.store_hot_state_diffs(&state_root, &state, &mut ops)?;
+                        db.hot_db.do_atomically(ops)?;
+                        diffs_written += 1;
+                    }
+                    StorageStrategy::ReplayFrom(_) => {
+                        // Optimization: instead of having to load the state of each summary we load x32
+                        // less states by manually computing the HotStateSummary roots using the
+                        // computed state dag.
+                        //
+                        // No need to store diffs for states that will be reconstructed by replaying
+                        // blocks.
+                        //
+                        // 2. Convert the summary to the new format.
+                        let previous_state_root = if state_root == split.state_root {
+                            Hash256::ZERO
```

Comment:
I think this is unreachable (could be an error), because the `split.state_root` will never be stored as a `ReplayFrom`. The anchor slot is set to the split slot, so the split state will be stored as a snapshot.

`Error::SplitStateIsNotASnapshot`

Reply:
Fixed